servers/src/common/adapters.rs

```diff
@@ -369,6 +370,19 @@ impl p2p::ChainAdapter for NetToChainAdapter {
 			true
 		}
 	}
+
+	/// Specific tmp dir.
+	/// Normally it's ~/.grin/main/tmp for mainnet
+	/// or ~/.grin/floo/tmp for floonet
+	fn get_tmp_dir(&self) -> PathBuf {
```

Comment:
Why not delegate to the chain implementation here? Am I missing something?

Reply:
accept